src/gepetto_viewer_rerun/client.py

```diff
@@ -74,27 +74,93 @@ def createScene(self, sceneName: str):
         )
         logger.info(msg)
 
-    def _getSceneIndex(self, sceneName: str):
-        for index, scene in enumerate(self.sceneList):
+    def getWindowList(self):
+        return [scene.name for scene in self.windowList]
+
+    def getSceneList(self):
+        return [scene.name for scene in self.sceneList]
+
+    def getWindowID(self, name: str):
+        assert isinstance(name, str), "Parameter 'name' must be a string"
+
+        if name not in self.getWindowList():
+            logging.error(f"getWindowID(): Unknown windowName '{name}'.")
+            return ""
+        logger.info("getWindowID() only checks if window exists and return its name.")
+        return name
+
+    def getNodeList(self):
+        entitiesName = []
+        for entity_type in self.entityList:
+            for entity in entity_type:
+                entitiesName.append(entity.name)
+        return self.getWindowList() + self.getSceneList() + entitiesName
+
+    def nodeExists(self, nodeName: str):
+        assert isinstance(nodeName, str), "Parameter 'nodeName' must be a string"
+
+        return nodeName in self.getNodeList()
+
+    def _getScene(self, sceneName: str):
+        for scene in self.sceneList:
             if scene.name == sceneName:
-                return index
-        return -1
+                return scene
+        return None
+
+    def _getWindow(self, windowName: str):
+        for window in self.windowList:
+            if window.name == windowName:
+                return window
+        return None
```

Comment:
```suggestion
    def _getWindow(self, windowName: str) -> Window | None:
        for window in self.windowList:
            if window.name == windowName:
                return window
```
Returning None is the same as not returning.
Adding return type hints is a bit more explicit than having a `return None` extra line.

And same above. And in general, adding return type hints to other methods would be good.
And we might add some type checker in the tooling, but this is out of scope for this PR

Reply:
Yes, at some point I planned to specify all return hints for better understanding, so this might be the next PR.